# Assignment 2: Probability

Task 1: Simulating Coin Flips
Write a Python program that simulates flipping a fair coin 100 times. Calculate and print the probability of getting heads and tails.

Task 2: Probability of Even Numbers
Write a Python function that takes a list of numbers as input and returns the probability of selecting an even number from the list.

Task 3: Divisibility by 7
Create a Python program that generates a random number between 1 and 10,000. Calculate and print the probability of the number being divisible by 7.

Task 4: Ball Selection
Write a Python program that simulates randomly selecting a ball from a bag containing red, blue, and green balls. Calculate and print the probability of selecting a red ball.

Task 5: First Axiom of Probability
Write a Python program that demonstrates the first axiom of probability by calculating the probability of an event occurring between 0 and 1.

In this assignment, you'll delve into the realm of probability through practical simulations and calculations. These tasks will help you grasp the fundamental concepts of probability and understand how to calculate and interpret probabilities of various events.

Task 1: 
Write a Python program to simulate flipping a fair coin 100 times. Calculate and print the 
probability of getting heads and tails.

In [1]:
#This program uses the random.choice() function to simulate 100 
#coin flips, and then it calculates the number of heads and tails. Finally,
#it calculates and prints the probabilities of getting heads and tails
#based on the simulated coin flips.

import random

# Simulate flipping a fair coin 100 times
coin_flips = [random.choice(['Heads', 'Tails']) for _ in range(100)]

# Calculate the number of heads and tails
num_heads = coin_flips.count('Heads')
num_tails = coin_flips.count('Tails')

# Calculate the probabilities
probability_heads = num_heads / 100
probability_tails = num_tails / 100

# Print the results
print(f"Number of Heads: {num_heads}")
print(f"Number of Tails: {num_tails}")
print(f"Probability of Heads: {probability_heads:.2f}")
print(f"Probability of Tails: {probability_tails:.2f}")


Number of Heads: 55
Number of Tails: 45
Probability of Heads: 0.55
Probability of Tails: 0.45


Number of Heads: 55

The coin landed on 'Heads' 55 times out of 100 flips.
Number of Tails: 45

The coin landed on 'Tails' 45 times out of 100 flips.
Probability of Heads: 0.55

There's a 55% chance of getting 'Heads' when flipping the coin.
Probability of Tails: 0.45

There's a 45% chance of getting 'Tails' when flipping the coin.
In short, the coin flip simulation showed that 'Heads' occurred more often, with a probability of around 55%, while 'Tails' occurred with a probability of about 45%.

Task 2: 
Write a Python function that takes a list of numbers as input and returns the probability of 
selecting an even number from the list.

In [2]:
#In this code snippet, you have a list of numbers from 1 to 10. You count 
#the number of even numbers (those divisible by 2) using a generator
#expression with sum. Then, you calculate the probability of selecting 
#an even number by dividing the count of even numbers by the total
#number of numbers in the list. Finally, you print the calculated probability.

# List of numbers
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

# Count the number of even numbers
even_count = sum(1 for i in numbers if i % 2 == 0)

# Calculate the probability of selecting an even number
# Probability = Count of Even Numbers / Total Number of Numbers
prob_even = even_count / len(numbers)

# Print the calculated probability
print(prob_even)



0.5


The result 0.5 represents the probability of selecting an even number from the list of numbers provided. In other words, there is a 50% chance of choosing a number that is divisible by 2 from the list. This probability is obtained by dividing the count of even numbers (5 even numbers) by the total number of numbers in the list (10 numbers).

Task 3: 
Create a Python program that generates a random number between 1 and 10,000. Calculate 
and print the probability of the number being divisible by 7.

In [3]:
#generate a random number and calculate the probability of that number
#being divisible by 7. The probability is set to 1/7, which is approximately 0.142857.
#The random number and the calculated probability are then printed.
#The comment at the end clarifies that the calculated probability represents about 14.2%.



# Generate a random number between 1 and 10,000
random_num = random.randint(1, 10000)

# Calculate the probability of the number being divisible by 7
prob_1 = 1 / 7

# Print the random number and the calculated probability
print(random_num, prob_1)

8264 0.14285714285714285


The number 3088 is a random number that was generated from the range 1 to 10,000. The value 0.14285714285714285 represents the probability that this randomly chosen number is divisible by 7. In simpler terms, there's roughly a 14.29% chance that a random number between 1 and 10,000 will be evenly divisible by 7.

Task 4: 
Write a Python program that simulates randomly selecting a ball from a bag containing red, 
blue, and green balls. Calculate and print the probability of selecting a red ball.

In [4]:
#randomly select a color from the list using random.choice(). Then,
#calculate the probability of selecting a red ball by dividing the count
#of red balls by the total number of balls.

#the variable name colors is used consistently throughout the code,
#ensuring that the list of colors is properly referenced when calculating
#the probability of selecting a red ball.

# List of ball colors in the bag
colors = ['red', 'blue', 'green', 'red', 'blue', 'green']

# Simulate randomly selecting a color from the list
selected_color = random.choice(colors)

# Calculate the probability of selecting a red ball
prob_red = colors.count('red') / len(colors)

# Print the calculated probability (expected result: 0.5)
print(prob_red)


0.3333333333333333


The result 0.3333333333333333 represents the probability of selecting a red ball from the list of colors. In other words, there's roughly a 33.33% chance of randomly choosing a red ball from the list. This probability is calculated by dividing the count of red balls (2 red balls) by the total number of balls in the list (6 balls).

Task 5: 
Write a Python program that demonstrates the first axiom of probability by calculating the 
probability of an event occurring between 0 and 1

In [5]:
#use random.uniform(0, 1) to generate a random probability value between 0 and 1. 
#Then, you check if the generated probability value
#falls within the range of 0 to 1 using an if statement.



# Generate a random probability between 0 and 1
probability = random.uniform(0, 1)

# Check if the generated probability is between 0 and 1
if 0 <= probability <= 1:
    print(f"The generated probability {probability} satisfies the first axiom of probability.")
else:
    print(f"The generated probability {probability} does not satisfy the first axiom of probability.")



The generated probability 0.5591155852799045 satisfies the first axiom of probability.


The statement "The generated probability 0.5591155852799045 satisfies the first axiom of probability" means that the randomly generated probability value (0.5591155852799045) falls within the valid range of probabilities between 0 and 1. In other words, the generated probability is a non-negative value less than or equal to 1, which aligns with the first axiom of probability.